### PR TITLE
Better support for upload delete events

### PR DESCRIPTION
**Pull request by cableman**
https://jira.itkdev.dk/browse/DDBTEAM-461

The whole process have been test in hand with the queue command.

### CODE REVIEW

File: src/Event/VendorEvent.php [hunk 1]
```diff
@@ -42,6 +43,17 @@ public function __construct(string $type, array $identifiers, string $identifier
...
 
+    /**
+     * Change the current state.
+     *
+     * @param $newVendorState
+     *   Change the state to this state. Please use the static properties found in the VendorState class
+     */
+    public function changeVendorState($newVendorState): void
+    {
+        $this->type = $newVendorState;
```
Comment (tuj) [inline src/Event/VendorEvent.php:54]: type = state ??

File: src/Event/VendorEvent.php [hunk 2]
```diff
@@ -42,6 +43,17 @@ public function __construct(string $type, array $identifiers, string $identifier
         $this->vendorId = $vendorId;
     }
 
+    /**
+     * Change the current state.
+     *
+     * @param $newVendorState
```
Comment (tuj) [inline src/Event/VendorEvent.php:49]: missing type...

File: src/Event/VendorEvent.php [hunk 3]
```diff
@@ -42,6 +43,17 @@ public function __construct(string $type, array $identifiers, string $identifier
...
+     *   Change the state to this state. Please use the static properties found in the VendorState class
+     */
+    public function changeVendorState($newVendorState): void
```
Comment (tuj) [inline src/Event/VendorEvent.php:52]: Missing parameter type

File: src/Event/VendorEvent.php (hunk 2 above)
Comment (cableman) [reply src/Event/VendorEvent.php:49]: I'm not sure what the type should be as they come from const, which have not type?

File: src/Event/VendorEvent.php (hunk 3 above)
Comment (cableman) [reply src/Event/VendorEvent.php:52]: I'm not sure what the type should be as they come from const, which have not type?

File: src/Event/VendorEvent.php (hunk 1 above)
Comment (cableman) [reply src/Event/VendorEvent.php:54]: Yes, it's technically the state which is also the type of the event...

Comment (cableman) [reply src/Event/VendorEvent.php:54]: Fixed

File: src/Event/VendorEvent.php (hunk 2 above)
Comment (cableman) [reply src/Event/VendorEvent.php:49]: Fixed

File: src/Event/VendorEvent.php (hunk 3 above)
Comment (cableman) [reply src/Event/VendorEvent.php:52]: Fixed

**Pull request merged**